### PR TITLE
GitLab EveryAuth example

### PR DESCRIPTION
**Pull request by degrammer**

### CODE REVIEW

**Review by LizzParody — APPROVED** [review]

File: examples/gitlab/README.md [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+Install the [uuid](https://www.npmjs.com/package/uuid) package. Used to generate a unique identifier for the userId.
+
+```shell
+npm i uuid
+```
+Install the [cookie-session](https://www.npmjs.com/package/cookie-session) package to allow your application to establish a session (an HTTP-Only cookie) to store the authorizing user id.
+
+```shell
+npm i cookie-session
+```
```
Comment (bennbollay) [inline examples/gitlab/README.md:55]: All of this should be just part of the package.json and not called out separately.  The session stuff is just there, for example, as illustrative purposes.

File: examples/gitlab/README.md [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+   . . . render the data
+});
+```
+
+Now, we need to display the data. We will use a simple template engine called [pug](https://www.npmjs.com/package/pug), which allows us to quickly render an HTML page with the data returned from GitHub.
+
+Install the dependency and set it as the default view engine:
+
+```shell
+npm i pug
```
Comment (bennbollay) [inline examples/gitlab/README.md:119]: I want to see less "how to make this example from scratch yourself" content in this README and more "this is what it's doing and the one or two relevant pieces, with a screenshot".

Good illustration, though, that the blog post as whole isn't a good target here.

File: examples/gitlab/index.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
+const express = require('express');
+const everyauth = require('@fusebit/everyauth-express');
```
Comment (bennbollay) [inline examples/gitlab/index.js:2]: Pull this out into a space by itself, i.e.:
```
const express = require('express');
const { Gitlab } = require('@gitbeaker/node');
const { v4: uuidv4 } = require('uuid');
const cookieSession = require('cookie-session');

const everyauth = require('@fusebit/everyauth-express');

const app = express();
```
etc.  That visually highlights it.

File: examples/gitlab/index.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+// Get userId from the authorization redirect or via session if already authorized.
+const handleSession = (req, res, next) => {
+  if (req.query.userId) {
+    req.session.userId = req.query.userId;
+  }
+  if (!req.session.userId) {
+    return res.redirect('/');
+  }
+  return next();
+};
```
Comment (bennbollay) [inline examples/gitlab/index.js:19]: Rename to `validateSession`.

File: examples/gitlab/index.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+  }
+  res.redirect('/finished');
+});
+
+app.use(
+  '/authorize/:userId',
+  (req, res, next) => {
+    if (!req.params.userId) {
+      return res.redirect('/');
+    }
```
Comment (bennbollay) [inline examples/gitlab/index.js:42]: Will the url match w/o a userId?

File: examples/gitlab/README.md (hunk 1 above)
Comment (degrammer) [reply examples/gitlab/README.md]: Fixed

File: examples/gitlab/index.js (hunk 5 above)
Comment (degrammer) [reply examples/gitlab/index.js:43]: if a userId is not provided, it checks the session, if empty it initializes the authorization flow.

Comment (bennbollay) [reply examples/gitlab/index.js:43]: No, I mean that `/authorize/:userId` won't match an inbound url `/authorize`, so the check doesn't make a lot of sense, unless I'm reading it wrong.

File: examples/gitlab/package.json [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+  "description": "Use GitLab API with EveryAuth",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "keywords": [],
+  "author": "Fusebit, Inc.",
+  "license": "ISC",
+  "dependencies": {
+    "@fusebit/everyauth-express": "^1.0.3",
```
Comment (bennbollay) [inline examples/gitlab/package.json:13]: `*`

**Review by bennbollay — APPROVED** [review]

File: examples/gitlab/index.js (hunk 5 above)
Comment (degrammer) [reply examples/gitlab/index.js]: you're right, doesn't make sense at all, no idea what  I was thinking with that middleware :P , removed!

**Pull request merged**